public/pages/DetectorResults/components/DetectorState/DetectorUnknownState.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+import React from 'react';
+import { EuiButton, EuiEmptyPrompt, EuiIcon } from '@elastic/eui';
+import { Fragment } from 'react';
+import { DETECTOR_INIT_FAILURES } from '../../../DetectorDetail/utils/constants';
+
+export interface DetectorUnknownStateProps {
+  onSwitchToConfiguration(): void;
+  onStartDetector(): void;
+}
+
+export const DetectorUnknownState = (props: DetectorUnknownStateProps) => {
+  return (
+    <EuiEmptyPrompt
+      style={{ maxWidth: '75%' }}
+      title={
+        <div>
+          <EuiIcon type="alert" size="l" color="danger" />
+
+          <h2>The detector is in unknown state</h2>
```

Comment:
Should we add a new unknown state? Or just use initialization failutre?

Reply:
I add this just for worst case scenario. Theoretically user should **never** reach this state.

It can only be shown to user if there is bug in our code. It could be new edge case we fail to cover, or some logic bug which leads to unexpected state. Such issues could be related to initialization, or config update, or any other bug on AD Kibana side. That's why I don't use initialization failure.